{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# MAKE SURE FUSION.compute_weights.R IS IN YOUR PATH\n",
    "# FILL IN THESE PATHS\n",
    "GCTA=\"/cellar/users/jlz014/gcta_1.91.4beta/gcta64\"\n",
    "PLINK=\"<PATH TO PLINK>\"\n",
    "GEMMA=\"/cellar/users/jlz014/gemma/bin/gemma\"\n",
    "# ALTERNATIVELY: ENSURE THAT plink, gcta, gemma CAN BE CALLED FROM PATH AND REMOVE --PATH_* FLAGS BELOW\n",
    "# PATH TO DIRECTORY CONTAINING LDREF DATA (FROM FUSION WEBSITE or https://data.broadinstitute.org/alkesgroup/FUSION/LDREF.tar.bz2)\n",
    "LDREF=\"/cellar/users/jlz014/fusion_twas-master/LDREF\"\n",
    "# THIS IS USED TO RESTRICT INPUT SNPS TO REFERENCE IDS ONLY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# PATH TO GEUVADIS GENE EXPRESSION MATRIX:\n",
    "PRE_GEXP=\"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt\"\n",
    "# GEUVADIS DATA WAS DOWNLOADED FROM https://www.ebi.ac.uk/arrayexpress/experiments/E-GEUV-1/files/analysis_results/\n",
    "\n",
    "# PATH TO PREFIX FOR GEUVADIS GENOTYPES SPLIT BY CHROMOSOME\n",
    "# SUBSAMPLE THESE TO THE LDREF SNPS FOR EFFICIENCY\n",
    "PRE_GENO=\"1000G\"\n",
    "\n",
    "# PATH TO OUTPUT DIRECTORY (population-specific subdirs will be made)\n",
    "OUT_DIR=\"./WEIGHTS\"\n",
    "\n",
    "# ROWS IN THE MATRIX TO ANALYZE (FOR BATCHED RUNS)\n",
    "BATCH_START=1\n",
    "BATCH_END=10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- BEGIN SCRIPT:\n",
    "\n",
    "NR=\"${BATCH_START}_${BATCH_END}\"\n",
    "mkdir --parents tmp/$NR\n",
    "mkdir --parents hsq/$NR\n",
    "mkdir --parents out/$NR\n",
    "# THIS IS DIRECTORY WHERE THE OUTPUT WILL GO:\n",
    "# mkdir $OUT_DIR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000136237.12 7 21896763 22896763\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.7\n",
      "  --chr 7\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.7.bim\n",
      "  --from-bp 21896763\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.pheno\n",
      "  --to-bp 22896763\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "673 variants loaded from .bim file.\n",
      "489 people (0 males, 0 females, 489 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.nosex\n",
      ".\n",
      "344 phenotype values present after --pheno.\n",
      "--extract: 673 variants remaining.\n",
      "--keep: 344 people remaining.\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 344 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "673 variants and 344 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.bed\n",
      "+\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.bim\n",
      "+\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "./WEIGHTS/ALL/ALL.ENSG00000136237.12\n",
      "[1] \".hsq saved to ./WEIGHTS/ALL/ALL.ENSG00000136237.12.hsq\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.ALL.tmp\"\n",
      "[1] \"./WEIGHTS/ALL/ALL.ENSG00000136237.12.wgt.RDat\"\n",
      "[1] \"output saved to ./WEIGHTS/ALL/ALL.ENSG00000136237.12.wgt.RDat\"\n",
      "Append heritability output to hsq file\n",
      "done\n"
     ]
    }
   ],
   "source": [
    "# test first gene in $PRE_GEXP\n",
    "PARAM=`awk 'NR==7' $PRE_GEXP`\n",
    "\n",
    "# # Loop through each gene expression phenotype in the batch\n",
    "# cat $PRE_GEXP | awk -vs=$BATCH_START -ve=$BATCH_END 'NR > s && NR <= e' |  while read PARAM; do\n",
    "\n",
    "# Get the gene positions +/- 500kb (1MB window)\n",
    "CHR=`echo $PARAM | awk '{ print $3 }'`\n",
    "P0=`echo $PARAM | awk '{ print $4 - 0.5e6 }'`\n",
    "P1=`echo $PARAM | awk '{ print $4 + 0.5e6 }'`\n",
    "GNAME=`echo $PARAM | awk '{ print $1 }'`\n",
    "\n",
    "OUT=\"tmp/$NR/$PRE_GEXP.$GNAME\"\n",
    "echo $GNAME $CHR $P0 $P1\n",
    "\n",
    "# Pull out the current gene expression phenotype\n",
    "echo $PARAM | tr ' ' '\\n' | tail -n+5 | paste $PRE_GEXP.ID - > $OUT.pheno\n",
    "\n",
    "# Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
    "plink2 --bfile $LDREF/$PRE_GENO.EUR.$CHR --pheno $OUT.pheno --make-bed --out $OUT --keep $OUT.pheno --chr $CHR --from-bp $P0 --to-bp $P1 --extract $LDREF/1000G.EUR.$CHR.bim\n",
    "\n",
    "echo 'Process all samples together (for reference purposes only since this is multi-ethnic data)'\n",
    "# Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
    "mkdir $OUT_DIR/ALL\n",
    "FINAL_OUT=\"$OUT_DIR/ALL/ALL.$GNAME\"\n",
    "echo $FINAL_OUT\n",
    "\n",
    "Rscript FUSION.compute_weights.R --bfile $OUT --tmp $OUT.ALL.tmp --out $FINAL_OUT --verbose 0 --save_hsq --PATH_gcta $GCTA --PATH_gemma $GEMMA --models blup,lasso,top1,enet\n",
    "\n",
    "# ALTERNATIVELY ADD COVARIATES HERE USING THE --covar FLAG\n",
    "# MINIMAL COMMAND IS: `Rscript FUSION.compute_weights.R --bfile $OUT --tmp $OUT.$pop.tmp --out $FINAL_OUT`\n",
    "\n",
    "echo 'Append heritability output to hsq file'\n",
    "# Append heritability output to hsq file\n",
    "cat $FINAL_OUT.hsq >> hsq/$NR.hsq\n",
    "\n",
    "# echo 'clean up'\n",
    "# # Clean-up just in case\n",
    "# rm -f $FINAL_OUT.hsq $OUT.tmp.*\n",
    "\n",
    "# echo 'remove all intermediate files'\n",
    "# # Remove all intermediate files\n",
    "# rm $OUT.*\n",
    "\n",
    "echo 'done'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000152931.6 5 59283540 60283540\n",
      "Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000152931.6.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.5\n",
      "  --chr 5\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.5.bim\n",
      "  --from-bp 59283540\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000152931.6.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000152931.6\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000152931.6.pheno\n",
      "  --to-bp 60283540\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "316 variants loaded from .bim file.\n",
      "489 people (0 males, 0 females, 489 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000152931.6.nosex\n",
      ".\n",
      "344 phenotype values present after --pheno.\n",
      "--extract: 316 variants remaining.\n",
      "--keep: 344 people remaining.\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 344 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "316 variants and 344 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000152931.6.bed +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000152931.6.bim +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000152931.6.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000152931.6.tmp.hsq does not exist, likely GCTA could not converge, skipping gene\n",
      "Append heritability output to hsq file\n",
      "cat: ./WEIGHTS/ALL/ALL.ENSG00000152931.6.hsq: No such file or directory\n",
      "clean up\n",
      "remove all intermediate files\n",
      "next gene\n",
      "ENSG00000183696.9 7 47628225 48628225\n",
      "Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000183696.9.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.7\n",
      "  --chr 7\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.7.bim\n",
      "  --from-bp 47628225\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000183696.9.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000183696.9\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000183696.9.pheno\n",
      "  --to-bp 48628225\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "539 variants loaded from .bim file.\n",
      "489 people (0 males, 0 females, 489 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000183696.9.nosex\n",
      ".\n",
      "344 phenotype values present after --pheno.\n",
      "--extract: 539 variants remaining.\n",
      "--keep: 344 people remaining.\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 344 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "539 variants and 344 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000183696.9.bed +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000183696.9.bim +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000183696.9.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "[1] \".hsq saved to ./WEIGHTS/ALL/ALL.ENSG00000183696.9.hsq\"\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000183696.9.tmp : heritability -0.010824; LRT P-value 0.34268 : skipping gene\n",
      "Append heritability output to hsq file\n",
      "clean up\n",
      "remove all intermediate files\n",
      "next gene\n",
      "ENSG00000139269.2 12 57346106 58346106\n",
      "Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000139269.2.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.12\n",
      "  --chr 12\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.12.bim\n",
      "  --from-bp 57346106\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000139269.2.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000139269.2\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000139269.2.pheno\n",
      "  --to-bp 58346106\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "345 variants loaded from .bim file.\n",
      "489 people (0 males, 0 females, 489 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000139269.2.nosex\n",
      ".\n",
      "344 phenotype values present after --pheno.\n",
      "--extract: 345 variants remaining.\n",
      "--keep: 344 people remaining.\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 344 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "345 variants and 344 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000139269.2.bed +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000139269.2.bim +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000139269.2.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000139269.2.tmp.hsq does not exist, likely GCTA could not converge, skipping gene\n",
      "Append heritability output to hsq file\n",
      "cat: ./WEIGHTS/ALL/ALL.ENSG00000139269.2.hsq: No such file or directory\n",
      "clean up\n",
      "remove all intermediate files\n",
      "next gene\n",
      "ENSG00000169129.8 10 115664515 116664515\n",
      "Get the locus genotypes for all samples and set current gene expression as the phenotype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000169129.8.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.10\n",
      "  --chr 10\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.10.bim\n",
      "  --from-bp 115664515\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000169129.8.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000169129.8\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000169129.8.pheno\n",
      "  --to-bp 116664515\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "475 variants loaded from .bim file.\n",
      "489 people (0 males, 0 females, 489 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000169129.8.nosex\n",
      ".\n",
      "344 phenotype values present after --pheno.\n",
      "--extract: 475 variants remaining.\n",
      "--keep: 344 people remaining.\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 344 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "475 variants and 344 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000169129.8.bed +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000169129.8.bim +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000169129.8.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "[1] \".hsq saved to ./WEIGHTS/ALL/ALL.ENSG00000169129.8.hsq\"\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000169129.8.tmp : heritability 0.013148; LRT P-value 0.30803 : skipping gene\n",
      "Append heritability output to hsq file\n",
      "clean up\n",
      "remove all intermediate files\n",
      "next gene\n",
      "ENSG00000134602.11 X 130657293 131657293\n",
      "Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000134602.11.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.X\n",
      "  --chr X\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.X.bim\n",
      "  --from-bp 130657293\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000134602.11.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000134602.11\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000134602.11.pheno\n",
      "  --to-bp 131657293\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "Error: Failed to open\n",
      "/cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.X.bed.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "ERROR: tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000134602.11.bed input file does not exist\n",
      "Append heritability output to hsq file\n",
      "cat: ./WEIGHTS/ALL/ALL.ENSG00000134602.11.hsq: No such file or directory\n",
      "clean up\n",
      "remove all intermediate files\n",
      "next gene\n",
      "ENSG00000136237.12 7 21896763 22896763\n",
      "Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.7\n",
      "  --chr 7\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.7.bim\n",
      "  --from-bp 21896763\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.pheno\n",
      "  --to-bp 22896763\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "673 variants loaded from .bim file.\n",
      "489 people (0 males, 0 females, 489 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.nosex\n",
      ".\n",
      "344 phenotype values present after --pheno.\n",
      "--extract: 673 variants remaining.\n",
      "--keep: 344 people remaining.\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 344 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "673 variants and 344 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.bed\n",
      "+\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.bim\n",
      "+\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "[1] \".hsq saved to ./WEIGHTS/ALL/ALL.ENSG00000136237.12.hsq\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp.cv.BSLMM\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp\"\n",
      "[1] \"tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000136237.12.tmp\"\n",
      "[1] \"./WEIGHTS/ALL/ALL.ENSG00000136237.12.wgt.RDat\"\n",
      "[1] \"output saved to ./WEIGHTS/ALL/ALL.ENSG00000136237.12.wgt.RDat\"\n",
      "Append heritability output to hsq file\n",
      "clean up\n",
      "remove all intermediate files\n",
      "next gene\n",
      "ENSG00000259425.1 15 22596869 23596869\n",
      "Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000259425.1.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  --chr 15\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.15.bim\n",
      "  --from-bp 22596869\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000259425.1.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000259425.1\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000259425.1.pheno\n",
      "  --to-bp 23596869\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "176 variants loaded from .bim file.\n",
      "489 people (0 males, 0 females, 489 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000259425.1.nosex\n",
      ".\n",
      "344 phenotype values present after --pheno.\n",
      "--extract: 176 variants remaining.\n",
      "--keep: 344 people remaining.\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 344 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "176 variants and 344 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000259425.1.bed +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000259425.1.bim +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000259425.1.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000259425.1.tmp.hsq does not exist, likely GCTA could not converge, skipping gene\n",
      "Append heritability output to hsq file\n",
      "cat: ./WEIGHTS/ALL/ALL.ENSG00000259425.1.hsq: No such file or directory\n",
      "clean up\n",
      "remove all intermediate files\n",
      "next gene\n",
      "ENSG00000242284.2 X 134453994 135453994\n",
      "Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000242284.2.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.X\n",
      "  --chr X\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.X.bim\n",
      "  --from-bp 134453994\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000242284.2.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000242284.2\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000242284.2.pheno\n",
      "  --to-bp 135453994\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "Error: Failed to open\n",
      "/cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.X.bed.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "ERROR: tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000242284.2.bed input file does not exist\n",
      "Append heritability output to hsq file\n",
      "cat: ./WEIGHTS/ALL/ALL.ENSG00000242284.2.hsq: No such file or directory\n",
      "clean up\n",
      "remove all intermediate files\n",
      "next gene\n",
      "ENSG00000235027.1 11 1281578 2281578\n",
      "Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000235027.1.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.11\n",
      "  --chr 11\n",
      "  --extract /cellar/users/jlz014/fusion_twas-master/LDREF/1000G.EUR.11.bim\n",
      "  --from-bp 1281578\n",
      "  --keep tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000235027.1.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000235027.1\n",
      "  --pheno tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000235027.1.pheno\n",
      "  --to-bp 2281578\n",
      "\n",
      "257959 MB RAM detected; reserving 128979 MB for main workspace.\n",
      "452 variants loaded from .bim file.\n",
      "489 people (0 males, 0 females, 489 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000235027.1.nosex\n",
      ".\n",
      "344 phenotype values present after --pheno.\n",
      "--extract: 452 variants remaining.\n",
      "--keep: 344 people remaining.\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 344 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "452 variants and 344 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000235027.1.bed +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000235027.1.bim +\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000235027.1.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./WEIGHTS/ALL’: File exists\n",
      "[1] \".hsq saved to ./WEIGHTS/ALL/ALL.ENSG00000235027.1.hsq\"\n",
      "tmp/1_10/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.ENSG00000235027.1.tmp : heritability 0.01812; LRT P-value 0.24467 : skipping gene\n",
      "Append heritability output to hsq file\n",
      "clean up\n",
      "remove all intermediate files\n",
      "next gene\n"
     ]
    }
   ],
   "source": [
    "# Loop through each gene expression phenotype in the batch\n",
    "cat $PRE_GEXP | awk -vs=$BATCH_START -ve=$BATCH_END 'NR > s && NR <= e' |  while read PARAM; do\n",
    "\n",
    "# Get the gene positions +/- 500kb (1MB window)\n",
    "CHR=`echo $PARAM | awk '{ print $3 }'`\n",
    "P0=`echo $PARAM | awk '{ print $4 - 0.5e6 }'`\n",
    "P1=`echo $PARAM | awk '{ print $4 + 0.5e6 }'`\n",
    "GNAME=`echo $PARAM | awk '{ print $1 }'`\n",
    "\n",
    "OUT=\"tmp/$NR/$PRE_GEXP.$GNAME\"\n",
    "echo $GNAME $CHR $P0 $P1\n",
    "\n",
    "# Pull out the current gene expression phenotype\n",
    "echo $PARAM | tr ' ' '\\n' | tail -n+5 | paste $PRE_GEXP.ID - > $OUT.pheno\n",
    "\n",
    "echo 'Get the locus genotypes for all samples and set current gene expression as the phenotype'\n",
    "\n",
    "# Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
    "plink2 --bfile $LDREF/$PRE_GENO.EUR.$CHR --pheno $OUT.pheno --make-bed --out $OUT --keep $OUT.pheno --chr $CHR --from-bp $P0 --to-bp $P1 --extract $LDREF/1000G.EUR.$CHR.bim\n",
    "\n",
    "echo 'Process all samples together (for reference purposes only since this is multi-ethnic data)'\n",
    "# Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
    "mkdir $OUT_DIR/ALL\n",
    "FINAL_OUT=\"$OUT_DIR/ALL/ALL.$GNAME\"\n",
    "\n",
    "# remove PATH_* flags because GCTA/GEMMA/PLINK2 can be called from path\n",
    "Rscript FUSION.compute_weights.R --bfile $OUT --tmp $OUT.tmp --out $FINAL_OUT --verbose 0 --save_hsq --PATH_gcta $GCTA --PATH_gemma $GEMMA --models blup,lasso,top1,enet\n",
    "\n",
    "# ALTERNATIVELY ADD COVARIATES HERE USING THE --covar FLAG\n",
    "# MINIMAL COMMAND IS: `Rscript FUSION.compute_weights.R --bfile $OUT --tmp $OUT.$pop.tmp --out $FINAL_OUT`\n",
    "\n",
    "echo 'Append heritability output to hsq file'\n",
    "# Append heritability output to hsq file\n",
    "cat $FINAL_OUT.hsq >> hsq/$NR.hsq\n",
    "\n",
    "echo 'clean up'\n",
    "# Clean-up just in case\n",
    "rm -f $FINAL_OUT.hsq $OUT.tmp.*\n",
    "\n",
    "echo 'remove all intermediate files'\n",
    "# Remove all intermediate files\n",
    "rm $OUT.*\n",
    "\n",
    "echo 'next gene'\n",
    "# GO TO THE NEXT GENE\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Process each population\n",
    "# THIS REQUIRES HAVING A $pop.keep FILE LISTING IDS FOR EACH POPULATION\n",
    "for pop in EUR YRI TSI CEU FIN GBR; do\n",
    "\tmkdir --parents WEIGHTS/$pop\n",
    "\tFINAL_OUT=\"WEIGHTS/$pop/$pop.$GNAME\"\n",
    "\t# EXTRACT THE POPULATION AND 1% SNPS\n",
    "\t$PLINK --bfile $OUT --keep $pop.keep --make-bed --out $OUT.$pop --maf 0.01\n",
    "\n",
    "\t# MAKE SURE FUSION.compute_weights.R IS IN YOUR PATH\n",
    "\tRscript FUSION.compute_weights.R --bfile $OUT.$pop --tmp $OUT.$pop.tmp --out $FINAL_OUT --verbose 0 --save_hsq --PATH_gcta $GCTA --PATH_gemma $GEMMA --models blup,lasso,top1,enet\n",
    "\t\n",
    "\t# Append heritability output to hsq file\n",
    "\tcat $FINAL_OUT.hsq >> hsq/$NR.hsq\n",
    "\n",
    "\t# Clean-up just in case\n",
    "\trm -f $FINAL_OUT.hsq $OUT.tmp.*\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
