{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# MAKE SURE FUSION.compute_weights.R IS IN YOUR PATH\n",
    "# FILL IN THESE PATHS\n",
    "GCTA=\"/cellar/users/jlz014/fusion_twas-master/gcta_nr_robust\"\n",
    "PLINK=\"<PATH TO PLINK>\"\n",
    "GEMMA=\"/cellar/users/jlz014/gemma/bin/gemma\"\n",
    "# ALTERNATIVELY: ENSURE THAT plink, gcta, gemma CAN BE CALLED FROM PATH AND REMOVE --PATH_* FLAGS BELOW\n",
    "\n",
    "# PATH TO DIRECTORY CONTAINING IMPUTED TCGA GENOTYPES\n",
    "TCGA_GENO='/cellar/users/jlz014/Data/TCGA/TCGA_EUR_imputed' \n",
    "# THIS IS USED TO RESTRICT INPUT SNPS TO REFERENCE IDS ONLY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# PATH TO GEUVADIS GENE EXPRESSION MATRIX:\n",
    "PRE_GEXP_DIR='/data/cellardata/users/jlz014/TCGA/expression/'\n",
    "PRE_GEXP=\"TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt\"\n",
    "\n",
    "# PATH TO PREFIX FOR GEUVADIS GENOTYPES SPLIT BY CHROMOSOME\n",
    "# SUBSAMPLE THESE TO THE LDREF SNPS FOR EFFICIENCY\n",
    "# test white population first\n",
    "PRE_GENO=\"EUR_imputed_females_filtered\"\n",
    "\n",
    "# PATH TO OUTPUT DIRECTORY (population-specific subdirs will be made)\n",
    "OUT_DIR=\"./TEST_WEIGHTS\"\n",
    "\n",
    "# THIS IS DIRECTORY WHERE THE OUTPUT WILL GO:\n",
    "if [[ ! -e $OUT_DIR ]]; then\n",
    "    mkdir -p $OUT_DIR\n",
    "fi\n",
    "\n",
    "# ROWS IN THE MATRIX TO ANALYZE (FOR BATCHED RUNS)\n",
    "BATCH_START=1\n",
    "BATCH_END=10\n",
    "\n",
    "# make *.ID for gene expression file\n",
    "echo `awk 'NR==1' $PRE_GEXP_DIR$PRE_GEXP` | tr ' ' '\\n' | tail -n+5 > $PRE_GEXP.ID.tmp\n",
    "paste $PRE_GEXP.ID.tmp $PRE_GEXP.ID.tmp | column -s $'\\t' -t > $PRE_GEXP.ID\n",
    "rm $PRE_GEXP.ID.tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make *.ID for gene expression file before script is submitted with the following commands:\n",
    "# THIS IS USED TO RESTRICT INPUT SNPS TO REFERENCE IDS ONLY\n",
    "PRE_GEXP_DIR='/data/cellardata/users/jlz014/TCGA/expression/'\n",
    "PRE_GEXP=\"TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt\"\n",
    "echo `awk 'NR==1' $PRE_GEXP_DIR/$PRE_GEXP` | tr ' ' '\\n' | tail -n+5 > $PRE_GEXP.ID.tmp\n",
    "paste $PRE_GEXP.ID.tmp $PRE_GEXP.ID.tmp | column -s $'\\t' -t > $PRE_GEXP.ID\n",
    "rm $PRE_GEXP.ID.tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-A8-A09Z  TCGA-A8-A09Z\n",
      "TCGA-5L-AAT1  TCGA-5L-AAT1\n",
      "TCGA-LL-A73Z  TCGA-LL-A73Z\n",
      "TCGA-AR-A2LQ  TCGA-AR-A2LQ\n",
      "TCGA-C8-A12X  TCGA-C8-A12X\n",
      "TCGA-A8-A081  TCGA-A8-A081\n",
      "TCGA-BH-A0W7  TCGA-BH-A0W7\n",
      "TCGA-S3-AA14  TCGA-S3-AA14\n",
      "TCGA-A1-A0SQ  TCGA-A1-A0SQ\n",
      "TCGA-W8-A86G  TCGA-W8-A86G\n"
     ]
    }
   ],
   "source": [
    "head $PRE_GEXP.ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000000419 20 49552684 49575069 -0.741658478192737 -0.0952878122051453 0.561424822441768 -1.30983090425029 -0.2322969207296 1.1635385691527 0.261360606434587 -1.21598010110204 1.71057139968799 -0.495239437590013 -0.75877860749 -0.124832641932182 -1.1635385691527 0.963054547438794 -0.22506250846222 0.380138579525698 -0.910233924905761 -2.11211410413361 -0.811537320762164 0.625875754409179 -0.868369424593738 -1.48414880852707 0.803400639985522 -1.04470907170884 0.0917485226718669 -0.732411362241843 -0.103550876090444 -2.23672082094666 -0.0587645509836792 -2.26634647382571 0.821372843922416 -0.16043320914408 1.55100820015369 0.131938669358162 1.90201162764356 0.438904105849384 0.0988282957743533 0.470163722234286 1.14524250362602 0.32265824618159 -0.18666758263932 -0.229884117579232 -0.572460100557655 -0.222653671572159 1.00693838147744 2.82785957952873 0.278415355587258 0.831288474199385 0.813171106895486 -2.7183696551449 -0.788886974508346 1.32663393933315 -0.258930522864627 -0.453176374508268 0.239543511284555 -0.141423810476247 0.215434847198847 2.9876164750834 2.0317959189859 -0.723226454407098 -0.676703351966606 -1.61687898783638 0.19623834359775 -1.59554488794423 -1.65262725374781 0.555932799787041 1.40418953263029 0.515257621466605 0.204627500458746 -0.263792234402169 -0.566933831484722 -0.0281944934711266 -0.75877860749 0.899620208156687 0.289421728574985 -0.307844626610119 -0.143797051777218 0.76978986132624 0.917367059418529 -1.16817385371869 1.97916383486611 1.47016055308686 0.332573274536095 -0.418301989104106 -0.0658259434697094 -1.23338528469306 1.25117235311421 -0.718656827729582 -0.763486377762817 -1.2409602565232 -1.86065524948111 0.480684999924859 -0.423435556965875 -1.634492039543 -0.834611881996458 0.464922608913104 0.370056750567784 1.77490377461874 0.614486190750816 1.01476983163407 -0.991457893403267 -0.283302834319547 0.258930522864627 2.07028683129411 0.309076473269985 0.0634717986465555 -0.0446501992265733 -0.273534518490431 2.20893723918423 2.63414828716326 -0.295551478359978 -1.09466259886652 1.23087596122945 0.0670031518826077 1.00110508381414 1.7312261676609 -0.546361826834125 0.340031007455912 0.931775282672411 -1.12732197883515 -1.10110064011295 -0.289421728574985 0.0328951498513163 -0.383929193959235 -0.946379585293759 0.584958762448342 -0.35000423395457 1.30983090425029 -1.31539069339733 0.158054002584512 -0.315242789044394 0.687077742159365 -0.821372843922416 -1.55100820015369 -0.238334882588802 -1.06313091448102 1.41369730623392 0.782489595129253 -0.878690917190458 0.854749601509769 -2.20893723918423 1.67132288064467 -0.596145862517023 0.578003907994439 -1.94690277624242 -0.738569093480834 1.16817385371869 -0.703531696048005 0.113003076413475 0.0941079185865668 -1.6906216295849 -0.837944533786756 -0.20822723215795 0.520630526051711 -1.36141265941811 -0.561424822441769 0.63447144590391 -0.85983833730774 0.445380245178927 0.185472456824871 1.01870898986592 -1.17987330353764 0.169959262448557 0.824669041130683 -0.0493538560632993 -0.968672678271665 -0.472789120992267 -0.054058605083459 0.714102158646556 -0.261360606434587 1.37640176419846 0.547726027908654 0.57384439721794 -0.254074923813032 0.427293061862712 -0.515257621466605 -0.298006478138073 -0.146171103260261 0.278415355587258 -0.115367671840826 0.179500764189909 0.244381546021853 0.839614353880728 0.177113890232518 -0.355003870440364 0.523322615263263 -0.28085825632883 0.534129257701477 -1.05489922843829 -0.887364203992537 -0.362520042480594 -0.383929193959235 -2.29810587751004 0.849682903970202 -0.397875921895433 0.955610650875884 0.273534518490431 0.950062296029712 1.15434223484386 -1.35550181110213 0.839614353880728 0.507226004042204 0.400419946518429 1.37640176419846 -0.844637925669689 0.135494160561078 0.135494160561078 0.422151123753597 0.227472651986704 0.834611881996458 -0.0493538560632993 0.497896916140058 1.49843365234221 -1.23338528469306 1.60399104625706 1.0467384093101 0.335057110390325 1.02464754491244 0.903146806526391 -1.11411555629751 -0.919157646018201 -1.27463987059825 -1.18697118199553 -0.0058731084465693 -0.987624756164258 0.59895427905732 -0.246802703048802 -0.36879918350899 0.66054606476184 -1.83475697814248 1.81003387454135 0.565554971864538 1.13176806106097 -1.67132288064467 0.139051365446838 0.72170157161161 0.604585346583237 -1.7208070198422 -0.136679702810405 -0.681140538958321 1.75264707546102 1.0861484760826 -0.648902923998513 -0.101189303831852 -0.377614537710531 0.935407648371426 -1.49483378873611 -0.73548674186458 1.2826203498285 1.78637004334039 0.117732912501695 -2.50677881437525 -0.357507012480026 -0.459694235725588 0.360012396568605 1.25632768492754 0.101189303831852 0.19145077100776 -0.0187949454787572 -0.509899550442469 0.491259760225725 -1.56684515283115 0.590543074029838 -0.117732912501695 -0.432446322354635 -1.04470907170884 -0.550457494448725 0.198633810817018 -0.121282013044516 0.237126601948541 -0.26622542312782 0.608821254074239 -1.41051387042409 -1.10541814661374 -0.162813324203774 0.90668467317363 -0.480684999924859 1.51303097881635 0.397875921895433 -1.0049901492137 0.0799589360347285 -1.78637004334039 1.44301611209358 -0.183082997942862 -1.02266401914355 1.02464754491244 -1.99609898948756 0.672279448524886 0.57384439721794 0.367542199426714 0.0611180055319083 0.942709665174611 -1.30431131138084 -0.515257621466605 -0.332573274536095 -0.935407648371426 -1.44970242840324 0.955610650875884 -0.365029970101007 -0.684106126179003 0.141423810476247 0.651805411229029 -1.29883117196802 -2.63414828716326 0.512576746510887 0.459694235725588 -1.33804743387177 0.0364211135176905 1.29068336351844 -0.485965587113828 -1.08192079778104 0.31771266237221 1.33804743387177 0.504556078314561 0.968672678271665 -1.01870898986592 1.6906216295849 1.65262725374781 0.640228022189158 -1.09466259886652 0.763486377762817 1.04268402734414 0.795316804415507 -0.256501967399465 1.03262228839855 -0.300463275326124 1.06106625139729 -0.134308808719688 2.01362675952574 -0.530069521317519 -0.0399475300462015 -0.468852238250719 -1.13399951454532 0.631601034954518 -0.372573643748843 1.1362366289844 -1.34382012373267 -0.951908491471257 -1.75264707546102 -0.669337477550217 1.52045215132091 -0.777712503742185 -0.285749106493027 -0.654713399971315 -1.14072800134343 0.72170157161161 0.0023492320339887 -0.47541778260874 0.509899550442469 -0.457084762813595 0.76663431186691 0.43631881069717 1.23590239854854 0.241961820714866 0.0517060875110767 -0.553193074095071 -0.462306842621128 0.252861963921735 0.337543015084535 0.457084762813595 0.213031083607043 -0.586353124326438 -0.818085546794524 0.565554971864538 0.798544075072335 -0.589145275516248 0.551824768081298 -1.55887780438092 -1.07561529539446 0.750969452310988 -0.0564114218729509 0.861539536031905 -0.183082997942862 -0.0305447372958899 2.3696083792069 0.920951184482767 0.126016532494062 0.325133993104783 0.268660189040486 1.03864671297483 0.987624756164258 -0.293098258588932 0.16043320914408 -1.13399951454532 -0.536840643436425 0.706543763943534 0.59895427905732 0.47541778260874 -0.939052398044821 0.648902923998513 -2.07028683129411 -1.91654925337734 1.82225620516861 0.252861963921735 1.27729099562854 1.12510727205236 1.70051191753956 0.483323608856113 -0.155675690374163 -1.52795801865074 -0.155675690374163 0.656169471635686 0.432446322354635 1.0861484760826 0.246802703048802 -0.687077742159365 -0.66054606476184 0.676703351966606 0.676703351966606 1.00110508381414 1.62562247712842 -2.0317959189859 1.49125319313136 -0.750969452310988 0.315242789044394 -0.148545978868403 -0.227472651986704 -1.58308513233281 0.772953094650171 0.365029970101007 -1.46327301059148 1.09252664154626 -1.06935227126506 1.06727389535369 -0.273534518490431 -0.150921692577794 -0.453176374508268 1.06106625139729 -0.708052203404029 -1.32099144298051 0.204627500458746 -0.957466650603694 -0.420867386615894 -1.39792086508513 2.18276092273369 0.813171106895486 -0.241961820714866 0.868369424593738 -0.167576336611829 3.30872838137925 -0.428580308232651 -0.426006523129839 -0.251649375945158 -1.6619024586693 1.26151662473702 -0.238334882588802 -0.523322615263263 1.55887780438092 -1.20619754364805 0.0564114218729509 -1.71057139968799 0.73548674186458 -1.97916383486611 0.883886911457566 -2.18276092273369 -0.555932799787041 -0.507226004042204 -1.33231899184227 -0.620170916280492 -1.90201162764356 -0.601767427642178 -1.50569221137522 0.472789120992267 0.201030418393635 0.0446501992265732 -0.326372613337757 1.35550181110213 -0.090569017035487 -0.0634717986465556 -0.17353546595522 -0.569694791233003 -1.20134927176049 0.0199697811409104 0.844637925669689 2.09074333855059 0.296778754662365 0.878690917190458 0.122465383916356 -0.387725333039388 2.26634647382571 -0.234711076984548 -1.52045215132091 0.19145077100776 -1.1206943350422 0.087031249394977 -0.0258444053726614 -1.42010751639822 -0.58078247689256 -0.216637195087369 -1.15434223484386 0.143797051777218 -0.86494943685256 -2.09074333855059 -0.352502950784336 1.39792086508513 -0.0164453503547771 -0.0858522377183731 -0.408067673041759 0.681140538958321 -1.68089400234473 -0.910233924905761 0.0422987477061101 0.889106879527235 -0.310308789115588 -0.445380245178926 -0.641670479900311 -3.30872838137928 0.22506250846222 -0.714102158646556 -0.564177186670628 -0.289421728574985 0.590543074029838 0.534129257701477 0.103550876090444 0.740112902854198 -0.432446322354635 -0.127200599706735 0.30538233195305 1.42983367683656 0.278415355587258 1.05081007100619 0.779302890990965 1.17283437435508 0.871799599349151 -0.0352457441483002 -1.15892815030076 1.09894956460625 -0.44019785324943 0.51794220475523 1.20134927176049 -1.28798638416694 -1.93150358830135 -0.63447144590391 1.44970242840324 2.56527929726853 2.11211410413361 -0.10945736341649 2.29810587751004 0.382665045452769 -0.495239437590013 0.58078247689256 -1.25117235311421 -0.542275303741502 1.12510727205236 0.40424085437968 1.07561529539446 -0.628735818524323 -1.02862673950904 0.892600357349046 -0.464922608913104 0.10945736341649 -1.82225620516861 0.181888661232992 -0.85983833730774 0.0199697811409104 -0.826320503576444 0.00469847703322272 -0.896104762929449 0.40424085437968 -1.27463987059825 0.10945736341649 -0.345013331076175 -0.663470833326644 -1.7312261676609 -0.625875754409179 0.75564941723939 -0.131938669358162 -0.0717129306221975 0.740112902854198 0.209427740274051 -2.01362675952574 -0.899620208156687 -2.45577519733961 -1.37337921080788 0.792097796021829 0.0587645509836792 -1.01084635694177 -0.320184475359531 -0.342521106440855 0.320184475359531 -0.178307200230425 -0.623020800800651 -0.578003907994439 1.47711855229621 1.07981419283685 -1.08826961272975 0.551824768081298 -0.926349667016646 -0.332573274536095 0.972435145963967 -0.534129257701477 0.45187511816982 -0.0952878122051453 -0.558676704738567 0.395334469750286 -1.22091530739026 -1.88786520522858 1.33231899184227 -1.06935227126506 1.60399104625706 0.656169471635686 -0.983806075379897 0.818085546794524 1.56684515283115 0.415739341602024 0.296778754662365 0.875240062589818 0.345013331076175 0.234711076984548 -0.195041032989808 1.87408679567777 -0.539555981576837 -0.0117464194853691 0.0305447372958901 -0.875240062589818 0.883886911457566 -0.641670479900311 0.976211430222823 -1.70051191753956 -0.693039264568071 -1.74183671811697 -0.105913025996777 -0.304151879521943 -0.798544075072335 0.664935347115724 -0.0352457441483002 -0.607408074539016 0.664935347115724 0.427293061862712 0.392795568503561 0.150921692577794 -0.980001687147238 0.864949436852561 -2.41045764637974 -0.402966565330928 -1.2933897637307 -1.43639369314091 -0.964923873535066 0.3029218874087 2.15799995815589 1.96277798010223 0.539555981576837 0.729342894187652 1.3917066541746 -1.38248488657413 0.355003870440364 0.534129257701477 2.45577519733961 -0.402966565330928 1.10975635801578 1.07143527669625 0.146171103260261 -0.727811233298492 -1.19412936914259 -0.976211430222823 0.293098258588932 0.604585346583237 -0.0717129306221975 -0.244381546021853 0.558676704738567 0.61165126880742 1.59136414942155 -0.340031007455912 0.357507012480026 -0.0752463031792155 -1.77490377461874 0.209427740274051 1.83475697814248 0.803400639985522 -0.483323608856113 -1.32663393933315 -0.278415355587258 -0.601767427642178 1.29068336351844 1.29883117196802 -0.0988282957743533 -1.03864671297483 0.732411362241843 -0.113003076413475 -1.53555096907914 -1.0049901492137 1.18697118199553 0.913794680236115 -0.216637195087369 0.0717129306221975 -0.910233924905761 0.0470018976587313 -0.814807066480889 1.34382012373267 -0.155675690374163 -1.79807610389363 0.219042832551728 1.93150358830135 0.17353546595522 1.32099144298051 0.711073927683066 0.418301989104106 -0.332573274536095 -2.82785957952873 0.229884117579232 -0.583565536947157 -1.02862673950904 0.717136952266521 -0.169959262448557 -0.961188580577452 1.03262228839855 -0.0611180055319083 0.995305653719743 1.19652907519603 1.09894956460625 -2.98761647508341 0.454478398648914 -0.84799884685904 1.24350113221773 -2.05066166875052 0.445380245178927 0.162813324203774 -0.414459042010561 0.981902104833956 1.86065524948111 -0.526018502532341 0.0823159120115245 2.7183696551449 0.827974222750931 0.413179421426693 1.46327301059148 -0.854749601509769 -0.205827114126704 0.375092899310459 0.478049733413089 0.385193956310167 0.289421728574985 0.17353546595522 -0.76978986132624 0.126016532494062 1.05489922843829 1.24350113221773 -1.54323349519663 1.99609898948756 -1.08192079778104 0.896104762929449 -0.76978986132624 -1.19412936914259 0.924547179270847 0.054058605083459 -0.711073927683066 -0.488610961802325 -1.14978046826562 -1.24605006148254 1.7208070198422 0.012921123835206 0.620170916280492 -0.747858594763302 -0.795316804415507 -1.62562247712842 -0.0140958460155701 0.467541560193239 1.48414880852707 0.485965587113828 1.36737145949952 -1.47016055308686 1.35550181110213 1.01084635694177 0.617326059815227 -0.273534518490431 2.33236781844344 -0.777712503742185 -0.495239437590013 -0.220246125938306 0.256501967399465 -0.63447144590391 -0.36879918350899 0.185472456824871 0.377614537710531 -0.090569017035487 -0.203428181192529 -0.744754957790462 -1.05900610158167 0.089389637388347 -0.501889744484826 0.167576336611829 0.351253318234326 1.6619024586693 -0.337543015084535 -1.60825738781759 0.462306842621128 -0.189058636141295 0.263792234402168 0.981902104833956 -0.504556078314561 0.15329825839831 0.726281277943541 -0.139051365446838 1.26673975516725 0.693039264568071 -1.38248488657413 -0.831288474199385 1.4564541919541 0.372573643748843 0.497896916140058 1.22339409931259 -0.43631881069717 -1.76366622231284 -0.871799599349151 -0.85136937427837 0.342521106440855 -0.268660189040486 1.84755122329497 0.669337477550217 0.623020800800651 -0.657626935805564 -1.10975635801578 -0.892600357349046 2.41045764637974 -1.84755122329497 -0.596145862517023 0.249225307700721 0.165194361951307 0.347507700500315 -0.377614537710531 -0.0493538560632993 0.702028053989228 2.50677881437525 0.584958762448342 -0.699025519860191 0.00704774796344828 -1.2589179177719 -0.0352457441483002 -1.87408679567777 1.79807610389363 -0.575229794397992 1.20619754364805 0.300463275326124 -0.17353546595522 1.35550181110213 -2.3696083792069 -2.33236781844344 -0.402966565330928 -0.591942027324369 -1.14524250362602 1.18223264692325 -0.696029274411311 0.0846733453702262 1.76366622231283 0.0670031518826077 -0.806648902221287 -0.841286518359547 0.441492337865043 1.19173651995533 -1.17283437435508 -0.942709665174611 -0.546361826834125 -0.445380245178926\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000000419 20 49052684 50075069\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/TCGA_EUR_imputed/EUR_imputed_females_filtered.20\n",
      "  --chr 20\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/TCGA_EUR_imputed/EUR_imputed_females_filtered.20.bim\n",
      "  --from-bp 49052684\n",
      "  --keep tmp/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.pheno\n",
      "  --make-bed\n",
      "  --out tmp/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419\n",
      "  --pheno tmp/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.pheno\n",
      "  --to-bp 50075069\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "2842 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "549 phenotype values present after --pheno.\n",
      "--extract: 2842 variants remaining.\n",
      "--keep: 549 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 549 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "2842 variants and 549 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.bed\n",
      "+\n",
      "tmp/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.bim\n",
      "+\n",
      "tmp/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "done\n"
     ]
    }
   ],
   "source": [
    "# TEST ONE \n",
    "# --- BEGIN SCRIPT:\n",
    "\n",
    "NR=\"${BATCH_START}_${BATCH_END}\"\n",
    "mkdir --parents tmp/$NR\n",
    "mkdir --parents hsq/$NR\n",
    "mkdir --parents out/$NR\n",
    "\n",
    "# test first gene in $PRE_GEXP\n",
    "PARAM=`awk 'NR==2' $PRE_GEXP_DIR$PRE_GEXP`\n",
    "echo $PARAM\n",
    "# Get the gene positions +/- 500kb (1MB window)\n",
    "CHR=`echo $PARAM | awk '{ print $2 }'`\n",
    "P0=`echo $PARAM | awk '{ print $3 - 0.5e6 }'`\n",
    "P1=`echo $PARAM | awk '{ print $4 + 0.5e6 }'`\n",
    "GNAME=`echo $PARAM | awk '{ print $1 }'`\n",
    "\n",
    "OUT=\"tmp/$PRE_GEXP.$GNAME\"\n",
    "echo $GNAME $CHR $P0 $P1\n",
    "\n",
    "# Pull out the current gene expression phenotype\n",
    "echo $PARAM | tr ' ' '\\n' | tail -n+5 | paste $PRE_GEXP.ID - > $OUT.pheno\n",
    "\n",
    "echo 'running plink'\n",
    "# # Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
    "plink2 --bfile $TCGA_GENO/$PRE_GENO.$CHR --pheno $OUT.pheno --make-bed --out $OUT --keep $OUT.pheno --chr $CHR --from-bp $P0 --to-bp $P1 --extract $TCGA_GENO/$PRE_GENO.$CHR.bim\n",
    "\n",
    "# echo 'Process all samples together (for reference purposes only since this is multi-ethnic data)'\n",
    "# # Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
    "# mkdir $OUT_DIR/ALL\n",
    "# FINAL_OUT=\"$OUT_DIR/ALL/ALL.$GNAME\"\n",
    "# echo $FINAL_OUT\n",
    "\n",
    "# Rscript FUSION.compute_weights.R --bfile $OUT --tmp $OUT.ALL.tmp --out $FINAL_OUT --verbose 1 --hsq_set 0.5 --PATH_gcta $GCTA --PATH_gemma $GEMMA --models blup,top1,enet\n",
    "\n",
    "# ALTERNATIVELY ADD COVARIATES HERE USING THE --covar FLAG\n",
    "# MINIMAL COMMAND IS: `Rscript FUSION.compute_weights.R --bfile $OUT --tmp $OUT.$pop.tmp --out $FINAL_OUT`\n",
    "\n",
    "# echo 'Append heritability output to hsq file'\n",
    "# # Append heritability output to hsq file\n",
    "# head $FINAL_OUT.hsq\n",
    "# cat $FINAL_OUT.hsq >> hsq/$NR.hsq\n",
    "\n",
    "# echo 'clean up'\n",
    "# # Clean-up just in case\n",
    "# rm -f $FINAL_OUT.hsq $OUT.tmp.*\n",
    "\n",
    "# echo 'remove all intermediate files'\n",
    "# # Remove all intermediate files\n",
    "# rm $OUT.*\n",
    "\n",
    "echo 'done'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘./TEST_WEIGHTS’: File exists\n",
      "ENSG00000000419 20 49052684 50075069\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.20\n",
      "  --chr 20\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.20.bim\n",
      "  --from-bp 49052684\n",
      "  --keep tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419\n",
      "  --pheno tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.pheno\n",
      "  --to-bp 50075069\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "2842 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "564 phenotype values present after --pheno.\n",
      "--extract: 2842 variants remaining.\n",
      "--keep: 564 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 564 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "2842 variants and 564 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.bed\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.bim\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000419.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./TEST_WEIGHTS/ALL’: File exists\n",
      "./TEST_WEIGHTS/ALL/ALL.ENSG00000000419\n",
      "[1] \"./TEST_WEIGHTS/ALL/ALL.ENSG00000000419.wgt.RDat\"\n",
      "[1] \"output saved to ./TEST_WEIGHTS/ALL/ALL.ENSG00000000419.wgt.RDat\"\n",
      "done\n",
      "ENSG00000000457 1 169328259 170363093\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000457.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1\n",
      "  --chr 1\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1.bim\n",
      "  --from-bp 169328259\n",
      "  --keep tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000457.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000457\n",
      "  --pheno tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000457.pheno\n",
      "  --to-bp 170363093\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "3470 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "564 phenotype values present after --pheno.\n",
      "--extract: 3470 variants remaining.\n",
      "--keep: 564 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 564 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "3470 variants and 564 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000457.bed\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000457.bim\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000457.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./TEST_WEIGHTS/ALL’: File exists\n",
      "./TEST_WEIGHTS/ALL/ALL.ENSG00000000457\n",
      "[1] \"./TEST_WEIGHTS/ALL/ALL.ENSG00000000457.wgt.RDat\"\n",
      "[1] \"output saved to ./TEST_WEIGHTS/ALL/ALL.ENSG00000000457.wgt.RDat\"\n",
      "done\n",
      "ENSG00000000460 1 169264549 170323221\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000460.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1\n",
      "  --chr 1\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1.bim\n",
      "  --from-bp 169264549\n",
      "  --keep tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000460.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000460\n",
      "  --pheno tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000460.pheno\n",
      "  --to-bp 170323221\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "3499 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "564 phenotype values present after --pheno.\n",
      "--extract: 3499 variants remaining.\n",
      "--keep: 564 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 564 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "3499 variants and 564 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000460.bed\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000460.bim\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000460.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./TEST_WEIGHTS/ALL’: File exists\n",
      "./TEST_WEIGHTS/ALL/ALL.ENSG00000000460\n",
      "[1] \"./TEST_WEIGHTS/ALL/ALL.ENSG00000000460.wgt.RDat\"\n",
      "[1] \"output saved to ./TEST_WEIGHTS/ALL/ALL.ENSG00000000460.wgt.RDat\"\n",
      "done\n",
      "ENSG00000000938 1 27449917 28452751\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000938.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1\n",
      "  --chr 1\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1.bim\n",
      "  --from-bp 27449917\n",
      "  --keep tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000938.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000938\n",
      "  --pheno tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000938.pheno\n",
      "  --to-bp 28452751\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "1695 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "564 phenotype values present after --pheno.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--extract: 1695 variants remaining.\n",
      "--keep: 564 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 564 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1695 variants and 564 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000938.bed\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000938.bim\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000938.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./TEST_WEIGHTS/ALL’: File exists\n",
      "./TEST_WEIGHTS/ALL/ALL.ENSG00000000938\n",
      "[1] \"./TEST_WEIGHTS/ALL/ALL.ENSG00000000938.wgt.RDat\"\n",
      "[1] \"output saved to ./TEST_WEIGHTS/ALL/ALL.ENSG00000000938.wgt.RDat\"\n",
      "done\n",
      "ENSG00000000971 1 196206037 197210216\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000971.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1\n",
      "  --chr 1\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1.bim\n",
      "  --from-bp 196206037\n",
      "  --keep tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000971.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000971\n",
      "  --pheno tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000971.pheno\n",
      "  --to-bp 197210216\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "1662 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "564 phenotype values present after --pheno.\n",
      "--extract: 1662 variants remaining.\n",
      "--keep: 564 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 564 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1662 variants and 564 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000971.bed\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000971.bim\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000000971.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./TEST_WEIGHTS/ALL’: File exists\n",
      "./TEST_WEIGHTS/ALL/ALL.ENSG00000000971\n",
      "[1] \"./TEST_WEIGHTS/ALL/ALL.ENSG00000000971.wgt.RDat\"\n",
      "[1] \"output saved to ./TEST_WEIGHTS/ALL/ALL.ENSG00000000971.wgt.RDat\"\n",
      "done\n",
      "ENSG00000001036 6 143323081 144332827\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001036.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.6\n",
      "  --chr 6\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.6.bim\n",
      "  --from-bp 143323081\n",
      "  --keep tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001036.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001036\n",
      "  --pheno tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001036.pheno\n",
      "  --to-bp 144332827\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "2501 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "564 phenotype values present after --pheno.\n",
      "--extract: 2501 variants remaining.\n",
      "--keep: 564 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 564 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "2501 variants and 564 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001036.bed\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001036.bim\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001036.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./TEST_WEIGHTS/ALL’: File exists\n",
      "./TEST_WEIGHTS/ALL/ALL.ENSG00000001036\n",
      "Error in reg$coef[2, 4] : subscript out of bounds\n",
      "Execution halted\n",
      "done\n",
      "ENSG00000001084 6 52879132 53888122\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001084.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.6\n",
      "  --chr 6\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.6.bim\n",
      "  --from-bp 52879132\n",
      "  --keep tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001084.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001084\n",
      "  --pheno tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001084.pheno\n",
      "  --to-bp 53888122\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "3087 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "564 phenotype values present after --pheno.\n",
      "--extract: 3087 variants remaining.\n",
      "--keep: 564 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 564 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "3087 variants and 564 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001084.bed\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001084.bim\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001084.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./TEST_WEIGHTS/ALL’: File exists\n",
      "./TEST_WEIGHTS/ALL/ALL.ENSG00000001084\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"./TEST_WEIGHTS/ALL/ALL.ENSG00000001084.wgt.RDat\"\n",
      "[1] \"output saved to ./TEST_WEIGHTS/ALL/ALL.ENSG00000001084.wgt.RDat\"\n",
      "done\n",
      "ENSG00000001167 6 40540721 41565689\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001167.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.6\n",
      "  --chr 6\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.6.bim\n",
      "  --from-bp 40540721\n",
      "  --keep tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001167.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001167\n",
      "  --pheno tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001167.pheno\n",
      "  --to-bp 41565689\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "3163 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "564 phenotype values present after --pheno.\n",
      "--extract: 3163 variants remaining.\n",
      "--keep: 564 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 564 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "3163 variants and 564 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001167.bed\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001167.bim\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001167.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./TEST_WEIGHTS/ALL’: File exists\n",
      "./TEST_WEIGHTS/ALL/ALL.ENSG00000001167\n",
      "[1] \"./TEST_WEIGHTS/ALL/ALL.ENSG00000001167.wgt.RDat\"\n",
      "[1] \"output saved to ./TEST_WEIGHTS/ALL/ALL.ENSG00000001167.wgt.RDat\"\n",
      "done\n",
      "ENSG00000001460 1 24241400 25243085\n",
      "running plink\n",
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001460.log.\n",
      "Options in effect:\n",
      "  --bfile /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1\n",
      "  --chr 1\n",
      "  --extract /cellar/users/jlz014/Data/TCGA/imputed/split_chr/EUR_imputed_females_filtered.1.bim\n",
      "  --from-bp 24241400\n",
      "  --keep tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001460.pheno\n",
      "  --make-bed\n",
      "  --out tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001460\n",
      "  --pheno tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001460.pheno\n",
      "  --to-bp 25243085\n",
      "\n",
      "249894 MB RAM detected; reserving 124947 MB for main workspace.\n",
      "2492 variants loaded from .bim file.\n",
      "3799 people (0 males, 3799 females) loaded from .fam.\n",
      "564 phenotype values present after --pheno.\n",
      "--extract: 2492 variants remaining.\n",
      "--keep: 564 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 564 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "2492 variants and 564 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001460.bed\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001460.bim\n",
      "+\n",
      "tmp/1_10/TCGA-BRCA.ENSG.chr1-22.median-filt.log2RPKM.qnorm.train.txt.ENSG00000001460.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
      "mkdir: cannot create directory ‘./TEST_WEIGHTS/ALL’: File exists\n",
      "./TEST_WEIGHTS/ALL/ALL.ENSG00000001460\n",
      "[1] \"./TEST_WEIGHTS/ALL/ALL.ENSG00000001460.wgt.RDat\"\n",
      "[1] \"output saved to ./TEST_WEIGHTS/ALL/ALL.ENSG00000001460.wgt.RDat\"\n",
      "done\n"
     ]
    }
   ],
   "source": [
    "# --- BEGIN SCRIPT:\n",
    "\n",
    "NR=\"${BATCH_START}_${BATCH_END}\"\n",
    "mkdir --parents tmp/$NR\n",
    "mkdir --parents hsq/$NR\n",
    "mkdir --parents out/$NR\n",
    "# THIS IS DIRECTORY WHERE THE OUTPUT WILL GO:\n",
    "mkdir $OUT_DIR\n",
    "\n",
    "# Loop through each gene expression phenotype in the batch\n",
    "cat $PRE_GEXP_DIR$PRE_GEXP | awk -vs=$BATCH_START -ve=$BATCH_END 'NR > s && NR <= e' |  while read PARAM; do\n",
    "\n",
    "# Get the gene positions +/- 500kb (1MB window)\n",
    "CHR=`echo $PARAM | awk '{ print $2 }'`\n",
    "P0=`echo $PARAM | awk '{ print $3 - 0.5e6 }'`\n",
    "P1=`echo $PARAM | awk '{ print $4 + 0.5e6 }'`\n",
    "GNAME=`echo $PARAM | awk '{ print $1 }'`\n",
    "\n",
    "OUT=\"tmp/$NR/$PRE_GEXP.$GNAME\"\n",
    "echo $GNAME $CHR $P0 $P1\n",
    "\n",
    "# Pull out the current gene expression phenotype\n",
    "echo $PARAM | tr ' ' '\\n' | tail -n+5 | paste $PRE_GEXP.ID - > $OUT.pheno\n",
    "\n",
    "echo 'running plink'\n",
    "# Get the locus genotypes for all samples and set current gene expression as the phenotype\n",
    "plink2 --bfile $TCGA_GENO/$PRE_GENO.$CHR --pheno $OUT.pheno --make-bed --out $OUT --keep $OUT.pheno --chr $CHR --from-bp $P0 --to-bp $P1 --extract $TCGA_GENO/$PRE_GENO.$CHR.bim\n",
    "\n",
    "echo 'Process all samples together (for reference purposes only since this is multi-ethnic data)'\n",
    "# Process all samples together (for reference purposes only since this is multi-ethnic data)\n",
    "mkdir $OUT_DIR/ALL\n",
    "FINAL_OUT=\"$OUT_DIR/ALL/ALL.$GNAME\"\n",
    "echo $FINAL_OUT\n",
    "\n",
    "Rscript FUSION.compute_weights.R --bfile $OUT --tmp $OUT.ALL.tmp --out $FINAL_OUT --verbose 0 --hsq_set 0.098 --PATH_gcta $GCTA --PATH_gemma $GEMMA --models top1,enet --noclean\n",
    "\n",
    "# ALTERNATIVELY ADD COVARIATES HERE USING THE --covar FLAG\n",
    "# MINIMAL COMMAND IS: `Rscript FUSION.compute_weights.R --bfile $OUT --tmp $OUT.$pop.tmp --out $FINAL_OUT`\n",
    "\n",
    "# echo 'Append heritability output to hsq file'\n",
    "# # Append heritability output to hsq file\n",
    "# cat $FINAL_OUT.hsq >> hsq/$NR.hsq\n",
    "\n",
    "# echo 'clean up'\n",
    "# # Clean-up just in case\n",
    "# rm -f $FINAL_OUT.hsq $OUT.tmp.*\n",
    "\n",
    "# echo 'remove all intermediate files'\n",
    "# # Remove all intermediate files\n",
    "# rm $OUT.*\n",
    "\n",
    "echo 'done'\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/cellar/users/jlz014/fusion_twas-master\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
